akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -691,6 +690,29 @@ object Source {
     new Source(scaladsl.Source.zipWithN[T, O](seq => zipper.apply(seq.asJava))(seq))
   }
 
+  /**
+   * Creates a `Source` that is materialized as an [[akka.stream.BoundedQueueSource]].
+   * You can push elements to the queue and they will be emitted to the stream if there is demand from downstream,
+   * otherwise they will be buffered until request for demand is received. The buffer size is passed in as a parameter.
+   * Elements in the buffer will be discarded if downstream is terminated.
+   *
+   * Pushed elements may be dropped if there is no space available in the buffer. Elements will also be dropped if the
+   * queue is completed or failed or if the stream completes.
```

Comment:
"queue is completed" - Why would that happen if the queue is completed? What does that mean?

"stream completes" - What does that mean?

E.g. clearer like this:

```suggestion
   * queue is failed through the materialized `BoundedQueueSource` or the source is cancelled by the downstream.
```

Reply:
SGTM